pkg/hubble/metrics/api/api.go

```diff
@@ -73,6 +75,9 @@ type Handler interface {
 	// specifies Prometheus registry
 	Init(registry *prometheus.Registry, options Options) error
 
+	// ProcessPodDeletion handles removing old metrics associated with deleted pod.
+	ProcessPodDeletion(ctx context.Context, podName, namespace string) error
```

Comment:
I believe this is the wrong design. Instead we should have an interface to get a list of metricsVec's from a handler. Then we can loop over every handler, and every MetricVector for each handler, calling `DeleteMetricsAssociatedWithPod`, instead of making every handler handle this explicitly.

Reply:
Refactored in a way you suggested.